src/esockd.erl

```diff
@@ -274,8 +275,17 @@ get_options({Proto, ListenOn}) when is_atom(Proto) ->
 %% because they require a listener restart, function returns an error.
 -spec set_options({atom(), listen_on()}, options()) ->
     ok | {error, not_supported | _UpdateErrorReason}.
-set_options({Proto, ListenOn}, Options) when is_atom(Proto) ->
-    with_listener_ref({Proto, ListenOn}, ?FUNCTION_NAME, [Options]).
+set_options({Proto, _ListenOn} = ListenerRef, Options) when is_atom(Proto) ->
+    OptionsWas = get_options(ListenerRef),
+    OptionsMerged = merge_opts(OptionsWas, Options),
+    with_listener_ref(ListenerRef, ?FUNCTION_NAME, [OptionsMerged]).
+
+%% @doc Replace set of applicable options
```

Comment:
From the first glance, this doesn't comment is not very descriptive. What does "reset" mean? Merge given options with the defaults?

Reply:
See refined comments.